### ISSUE TITLE
Changes to CMakeLists.txt are not detected

### ISSUE THREAD

**Issue by l0calh05t** [inbilla/CMake#16]
Steps to reproduce:
- Create a CMake project
- Configure using cmake (and FASTBuild generator)
- Optional: Build project
- Make changes to CMakeLists.txt and source files so that build fails if no reconfigure occurs
- Build project

Result:

Build fails, as fbuild.bff is not updated.

**Comment by inbilla** [comment]
Agreed - currently there is no detection for a CMakeList.txt file that is out of date.
I don't believe this is a feature supported by all CMake generators - I'm not sure when fastbuild generator would be able to support this, as fastbuild would need to be able to allow it's current script file to be reloaded/restarted.

**Comment by l0calh05t** [comment]
So far this is the first one I have encountered with this behavior. It works with Makefiles, MinGW Makefiles, NMake, all VS generators from 2008-2015 and Ninja. The VS generators typically don't reload automatically, so that isn't really an issue, but it would of course be nice.
